### ISSUE TITLE
Implement HasStorage for Executable

### ISSUE THREAD

**Issue by pmrv** [pyiron/pyiron_base#726]
We have a few places now where information about the executable needs to survive reloading of the job e.g. whether it's an [MPI executable](https://github.com/pyiron/pyiron_base/pull/713) or what the [accepted return codes](https://github.com/pyiron/pyiron_base/pull/537) are (currently broken because of this).

In the pyiron meeting we discussed to solve this by 
- [x] deriving `Executable` from `HasStorage`, then
- [x] storing everything in the `DataContainer`
- [ ] saving and restoring the `Executable` when saving and restoring `GenericJob`

**Comment by Leimeroth** [comment]
As discussed in there I tested #728 with my remote Setup for a simple VASP job. Back then everything seemed to work, but know I got an error on the cluster because the path to the atomicrex executable was a hardcoded version to the path on my desktop PC
after switching the branch on the cluster from #702 to master.

Right now I am not sure anymore if I changed the branch on both our cluster and my PC or just on the PC when I tested it, so maybe this caused the problem to only show up now. For now I switched back to #702 on the cluster and everything seems to work again, I just fear the point at which incompatibilities between branches break my system.
